js/exporter/jspdf/export.js

```diff
@@ -108,37 +114,47 @@ export const Export = {
         });
     },
 
-    assignCellStyle: function(pdfCell, gridCell, column, cellStyle) {
-        if(gridCell.rowType === 'header') {
-            // eslint-disable-next-line spellcheck/spell-checker
-            if(column.alignment) { pdfCell.styles.halign = column.alignment; }
-        } else {
-            if(cellStyle) {
-                // eslint-disable-next-line spellcheck/spell-checker
-                if(cellStyle.alignment) { pdfCell.styles.halign = cellStyle.alignment; }
-                if(cellStyle.bold) { pdfCell.styles.fontStyle = 'bold'; }
-                if(cellStyle.wrapText) { pdfCell.styles.cellWidth = 'wrap'; }
-                if(cellStyle.dataType === 'date') { pdfCell.content = dateLocalization.format(new Date(pdfCell.content), cellStyle.format); }
+    _getFormattedValue: function(value, format) {
+        if(isDefined(format)) {
+            if(isDate(value)) {
+                return dateLocalization.format(value, format);
+            }
+            if(isNumeric(value)) {
+                return numberLocalization.format(value, format);
             }
         }
+        return value;
     },
 
-    getDefaultPdfColumnWidths(autoTableWidth, columnWidths) {
-        if(!isNumeric(autoTableWidth) || !isDefined(columnWidths)) {
-            return;
+    _getPDFCellStyles: function(rowType, columnAlignment, cellStyle) {
+        const { alignment: horizontalAlignment, bold, wrapText } = cellStyle;
+        const styles = {};
+        const setStyle = (styleName, value) => {
+            if(isDefined(value)) {
+                styles[styleName] = value;
+            }
+        };
+        if(rowType === 'header') {
+            setStyle('halign', columnAlignment);
+        } else {
+            setStyle('halign', horizontalAlignment);
+            setStyle('fontStyle', bold ? 'bold' : undefined);
```

Comment:
Why do you use a condition? If there is a check on the undefined value inside the setStyle method.

Reply:
this block of code is rewritten